### PR TITLE
Corrected date format issue for different locale

### LINKED ISSUE #24696: Invalid input datetime format of value in Others locale 2.3.2
### Preconditions (*)

1.Magento 2.3.2 & 2.3-develop
2.Japen or China locale in admin 
* Backend-> Admin -> change the Backend admin panel language to Chinese

### Steps to reproduce (*)

1.Create or edit product
2.Change date field for new from to anything
3.Click Save

### Expected result (*)

1. Product Saves with new from date

### Actual result (*)

1.Error occurs stating "Invalid input datetime format of value '2019/9/24'
image

If choose EN is working, but in others locale, like JP showing chinese will be error.

### Additional information
https://github.com/magento/magento2/issues/24696#issuecomment-534514561

> Not a local options,
> Please follow this step ;
> Backend-> Admin -> change the Backend admin panel language to Chinese (Taiwan,hongkong,China) or Japanese and then go to product > add or edit product > add a “special price or new product”date time, it will be showing “xxxx年xx月xx日” not a xxxx/xx/xx.

### PR DESCRIPTION
**Pull request by Digisha-97**
Invalid Date Time Format issue for other locale
parterres-contribution  Partner: CommercePundit

### Description (*)

Date Time format is working properly for US only but when we change it to Germany or any other locale. Its throwing error on products save when we are adding values for  **Set Product as New From**. 

Reason it is not changing date format according configured locale.

Currently added solution for it, so now when we will change locale from configuration it will work fine.

Changed the filter method, as here system is not getting proper value to change according to selected locale.

 `$value = $this->_localeDate->formatDateTime($value, \IntlDateFormatter::SHORT, \IntlDateFormatter::SHORT, $currentLocaleCode, null, null);`

the value in filter method was not coming proper for other locale. so updated it as above. 

I have checked this fix on **Magento 2.3.2 and 2.3.3**

### Fixed Issues (if relevant)

1. fixes magento/magento2#24696: Invalid input datetime format of value in Others locale 2.3.2 #24696

### Manual testing scenarios (*)

1. Change locale to Germany from admin  > Stores > Configuration > General > Locale Options 
    here change **Locale** option to Germany
2. On top right corner admin menu click on Account Settings(admin)
    here change **Interface Locale** to Germany 
3. Navigate to Catalog  > Products 
    Select any product to edit it
    here add value for **Set Product as New From** for product and click on Save button

### Questions or comments

This is my first contribution , so please let me know if I missed anything or any correction require from my  #side.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by slavvka** [comment]
@magento run all tests

**Comment by cscd98** [comment]
Can this be merged please??

**Comment by Digisha-97** [comment]
I am not able to merge it. Merging is blocked.

Regards,

<https://www.commercepundit.com/>

*Digisha Patel, Development Technical Lead*
(O) 079-40399638 / 39
digisha@commercepundit.com

File: lib/internal/Magento/Framework/Stdlib/DateTime/Filter/DateTime.php [hunk 1]
```diff
@@ -12,13 +13,19 @@
...
+     * @var Resolver
+    */
+    private $localeResolver;
+
     /**
      * @param \Magento\Framework\Stdlib\DateTime\TimezoneInterface $localeDate
      *
      */
-    public function __construct(\Magento\Framework\Stdlib\DateTime\TimezoneInterface $localeDate)
+    public function __construct(\Magento\Framework\Stdlib\DateTime\TimezoneInterface $localeDate, Resolver $localeResolver)
```
Comment (sidolov) [inline lib/internal/Magento/Framework/Stdlib/DateTime/Filter/DateTime.php]: Please, inject new dependencies according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.4/contributor-guide/backward-compatible-development/)

**Review by sidolov — CHANGES_REQUESTED** [review]
@Digisha-97 please, cover changes with automated tests

File: lib/internal/Magento/Framework/Stdlib/DateTime/Filter/DateTime.php (hunk 1 above)
Comment (Digisha-97) [reply lib/internal/Magento/Framework/Stdlib/DateTime/Filter/DateTime.php]: public function __construct(\Magento\Framework\Stdlib\DateTime\TimezoneInterface $localeDate, ?Resolver $localeResolver = null) 
    {
        parent::__construct($localeDate);
        $this->localeResolver = $localeResolver;
     
        $this->_localToNormalFilter = $_localToNormalFilter ?: ObjectManager::getInstance()->create(Resolver::class);

        
[DateTime.zip](https://github.com/magento/magento2/files/5112348/DateTime.zip)
Hi ,

I have updated it. 
but I am not able to push my changes. 
getting below error :
warning: url has no scheme: github.com
fatal: credential url cannot be parsed: github.com

**Comment by engcom-Echo** [comment]
I will take care of test coverage.

**Comment by engcom-Echo** [comment]
@magento run all tests

**Review by sidolov — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-8133](https://jira.corp.magento.com/browse/ENGCOM-8133) has been created to process this Pull Request
:eight_spoked_asterisk: @sidolov, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Echo** [comment]
Hi, @Digisha-97. Continuing to investigate, unfortunately, I could not reproduce the issue on 2.4-develop.
Look to file "lib/internal/Magento/Framework/Stdlib/DateTime/Timezone.php".
image
Here we already get the current locale. In my opinion, this will be enough.
Instead, using "$ this -> _ localeDate-> formatDateTime ()" will fail, for example, when "$value = 28/11/2020".
image

**Review by engcom-Alfa — CHANGES_REQUESTED** [review]
Hi @Digisha-97 . 

Unfortunately, we are not able to reproduce this issue on fresh 2.4-develop

**Manual testing scenario:**

1. Change locale to Germany from **admin > Stores > Configuration > General > Locale Options**
and change **Locale** option to **Germany**;
2. On the top right corner admin menu click on Account Settings(admin)
and change Interface Locale to Germany;
3. Navigate to **Catalog > Products;**
4. Select any product to edit it;
5. Add value for Set Product as New From for product and click on the Save button;

**Actual Result:** :heavy_check_mark:  The product with **New From** - **To** values was successfully saved

Peek 2020-09-15 11-42

@Digisha-97 Could you take a look?
Thanks!

**Comment by Digisha-97** [comment]
Hi @engcom-Alfa ,

Yes this issue is not coming in magento 2.4

**Comment by engcom-Echo** [comment]
Hi @Digisha-97 , I'm closing this PR we can't process your pull request.
Please reopen and update if you wish to continue.
Thank you for your collaboration

**Pull request closed without merging**